eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClientListWithNextLinkAsyncCollectionResult.cs

```diff
@@ -0,0 +1,90 @@
+        /// <param name="context"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        public BasicTypeSpecClientListWithNextLinkAsyncCollectionResult(BasicTypeSpecClient client, Uri nextPage, RequestContext context)
+        {
+            _client = client;
+            _nextPage = nextPage;
+            _context = context;
+        }
+
+        /// <summary> Gets the pages of BasicTypeSpecClientListWithNextLinkAsyncCollectionResult as an enumerable collection. </summary>
+        /// <param name="continuationToken"> A continuation token indicating where to resume paging. </param>
+        /// <param name="pageSizeHint"> The number of items per page. </param>
+        /// <returns> The pages of BasicTypeSpecClientListWithNextLinkAsyncCollectionResult as an enumerable collection. </returns>
+        public override async IAsyncEnumerable<Page<BinaryData>> AsPages(string continuationToken, int? pageSizeHint)
+        {
+            do
+            {
+                Response response = await GetNextResponse(pageSizeHint, continuationToken).ConfigureAwait(false);
+                if (response is null)
+                {
+                    yield break;
+                }
+                ListWithNextLinkResponse responseWithType = (ListWithNextLinkResponse)response;
+                List<BinaryData> items = new List<BinaryData>();
+                foreach (var item in responseWithType.Things)
+                {
+                    items.Add(BinaryData.FromObjectAsJson(item));
+                }
+                continuationToken = responseWithType.Next.AbsoluteUri;
+                yield return Page<BinaryData>.FromValues(items, continuationToken, response);
+            }
+            while (!string.IsNullOrEmpty(continuationToken));
+        }
+
+        /// <summary> Get response from next link. </summary>
+        /// <param name="pageSizeHint"> The number of items per page. </param>
+        /// <param name="continuationToken"> A continuation token indicating where to resume paging. </param>
+        private async ValueTask<Response> GetNextResponse(int? pageSizeHint, string continuationToken)
+        {
+            HttpMessage message = _client.CreateListWithNextLinkRequest(_nextPage, _context);
+            using DiagnosticScope scope = _client.ClientDiagnostics.CreateScope("BasicTypeSpecClient.ListWithNextLink");
+            scope.Start();
+            try
+            {
+                await _client.Pipeline.SendAsync(message, _context.CancellationToken).ConfigureAwait(false);
+                return GetResponse(message);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Get response from message. </summary>
+        /// <param name="message"> Http message. </param>
+        private Response GetResponse(HttpMessage message)
```

Comment:
nit: Is there any reason we didn't just inline this logic in GetNextResponse?

Reply:
since we separate sync and async methods, we can just inline this.